soseki/reader/tokenization.py

```diff
@@ -148,29 +119,30 @@ def _compute_best_answer_spans(
 
         return selected_spans
 
-    def _get_answer_passage_texts_from_input_ids(
+    def _get_input_and_answer_texts(
         self,
         input_ids: List[int],
-        attention_mask: List[int],
-        token_type_ids: List[int],
         answer_span: AnswerSpan,
-    ) -> Tuple[str, str]:
-        input_tokens = self.tokenizer.convert_ids_to_tokens(input_ids)
-
-        answer_tokens = input_tokens[answer_span.start : answer_span.end + 1]
+    ) -> Tuple[str, str, Tuple[int, int]]:
+        left_input_ids = input_ids[: answer_span.start]
+        answer_input_ids = input_ids[answer_span.start : answer_span.end + 1]
 
-        passage_start = token_type_ids.index(1)
-        passage_length = token_type_ids.count(1)
-        passage_tokens = input_tokens[passage_start : passage_start + passage_length]
+        input_tokens = self.tokenizer.convert_ids_to_tokens(input_ids)
+        left_input_tokens = self.tokenizer.convert_ids_to_tokens(left_input_ids)
+        answer_tokens = self.tokenizer.convert_ids_to_tokens(answer_input_ids)
 
+        input_text = self.tokenizer.convert_tokens_to_string(input_tokens)
+        left_input_text = self.tokenizer.convert_tokens_to_string(left_input_tokens)
         answer_text = self.tokenizer.convert_tokens_to_string(answer_tokens)
-        passage_text = self.tokenizer.convert_tokens_to_string(passage_tokens)
 
-        return answer_text, passage_text
+        answer_start = input_text.find(answer_text, len(left_input_text))
+        answer_end = answer_start + len(answer_text)
+
+        return input_text, answer_text, (answer_start, answer_end)
```

Comment:
`_get_input_and_answer_texts()` now returns the start / end positions of the `answer_text` in the `input_text`.